orchestrator/test_runner/src/cross_bridge_balances.rs

```diff
@@ -0,0 +1,760 @@
+use crate::airdrop_proposal::wait_for_proposals_to_execute;
+use crate::happy_path::test_valset_update;
+use crate::happy_path_v2::deploy_cosmos_representing_erc20_and_check_adoption;
+use crate::ibc_auto_forward::{get_channel_id, setup_gravity_auto_forwards};
+use crate::{
+    create_default_test_config, footoken_metadata, get_ibc_chain_id, one_eth, start_orchestrators,
+    submit_false_claims_results, vote_yes_on_proposals, CosmosAddress, EthPrivateKey,
+    GravityQueryClient, ValidatorKeys, ADDRESS_PREFIX, COSMOS_NODE_GRPC, GRAVITY_MODULE_ADDRESS,
+    IBC_ADDRESS_PREFIX, MINER_PRIVATE_KEY, OPERATION_TIMEOUT, STAKING_TOKEN,
+};
+use clarity::Address as EthAddress;
+use cosmos_gravity::proposals::{
+    submit_set_monitored_token_addresses_proposal, SetMonitoredTokenAddressesProposal,
+};
+use cosmos_gravity::query::get_gravity_params;
+use cosmos_gravity::send::send_to_eth;
+use cosmos_gravity::utils::get_gravity_monitored_erc20s;
+use deep_space::{Coin, Contact, CosmosPrivateKey, Fee, PrivateKey};
+use ethereum_gravity::send_to_cosmos::send_to_cosmos;
+use gravity_proto::cosmos_sdk_proto::cosmos::bank::v1beta1::Metadata;
+use gravity_proto::cosmos_sdk_proto::ibc::core::channel::v1::query_client::QueryClient as IbcChannelQueryClient;
+use gravity_proto::gravity::{query_client::QueryClient, Erc20Token};
+
+use crate::unhalt_bridge::get_nonces;
+use futures::future::join;
+use gravity_utils::num_conversion::{downcast_uint256, one_atom};
+use gravity_utils::types::{
+    BatchRelayingMode, BatchRequestMode, RelayerConfig, ValsetRelayingMode,
+};
+use num256::Uint256;
+use relayer::main_loop::single_relayer_iteration;
+use std::ops::Mul;
+use std::time::Duration;
+use tokio::time::sleep;
+use tonic::transport::Channel;
+use web30::client::Web3;
+use web30::types::SendTxOption;
+
+const FOOTOKEN_ALLOCATION: u64 = 100u64; // Validators will have 100 FOO to spend
+
+/// Creates bridge activity, sets the MonitoredTokenAddresses with the 3 deployed ERC20's + footoken
+/// then tries several ways of halting the chain before finally verifying that an incorrect balance
+/// successfully halts the chain
+///
+/// Note: Not idempotent - will fail on successive runs
+pub async fn cross_bridge_balance_test(
+    web30: &Web3,
+    grpc: QueryClient<Channel>,
+    contact: &Contact,
+    keys: Vec<ValidatorKeys>,
+    _ibc_keys: Vec<CosmosPrivateKey>,
+    gravity_address: EthAddress,
+    erc20_addresses: Vec<EthAddress>,
+) {
+    // Disable the relayer so that complicated state can exist without race conditions
+    let mut no_relayer_config = create_default_test_config();
```

Comment:
this may be covering up the previously mentioned gov vote changing the list mid attestation problem.